symfit/core/minimizers.py

```diff
@@ -111,6 +125,7 @@ def __init__(self, *args, **kwargs):
         self._pickle_kwargs['jacobian'] = self.jacobian
 
         if self.jacobian is not None:
+            self.jacobian = self._baseobjective_from_callable(self.jacobian)
```

Comment:
The Jacobian is not an objective is it? If it needs wrapping, Objective should do that I think.

Reply:
This is done in case a custom callable is given as a jacobian, just like for the objective. This happens for example when doing things like:
```python
def func2d_symfit(x1, x2):
    f = np.cos(14.5 * x1 - 0.3) + (x2 + 0.2) * x2 + (x1 + 0.2) * x1
    return f

def jac2d_symfit(x1, x2):
    df = np.zeros(2)
    df[0] = -14.5 * np.sin(14.5 * x1 - 0.3) + 2. * x1 + 0.2
    df[1] = 2. * x2 + 0.2
    return df

x1, x2 = parameters('x1, x2')
fit = BFGS(func2d_symfit, [x1, x2], jacobian=jac2d_symfit)
```

Reply:
Ok, good point.

Reply:
On second thought I still don't like it. People calling minimizers should do their own wrapping. This means either 1) creating a `CallableNumericalModel` which takes a `jacobian` argument, or 2) calling `Fit`, which should then also take a `jacobian` argument.